api/python/provisioner/salt_api/base.py

```diff
@@ -0,0 +1,296 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from abc import ABC, abstractmethod
+from typing import (
+    List, Union, Dict, Tuple, Any, Type
+)
+import logging
+
+from ..vendor import attr
+from ..config import (
+   SECRET_MASK
+)
+from ..errors import (
+    SaltNoReturnError,
+    SaltCmdRunError, SaltCmdResultError
+)
+
+logger = logging.getLogger(__name__)
+
+
+# TODO TEST
+@attr.s(auto_attribs=True)
+class SaltArgsBase:
+    _prvsnr_type_ = True
+
+    fun: str = attr.ib(converter=str)
+    fun_args: Tuple = attr.ib(
+        converter=lambda v: () if v is None else v, default=None
+    )
+    fun_kwargs: Dict = attr.ib(
+        converter=lambda v: {} if v is None else v, default=None
+    )
+    kw: Dict = attr.Factory(dict)
+    secure: bool = False
+
+    @property
+    def args(self):
+        return (self.fun,)
```

Comment:
I think, it will be more readable as:
```python
return self.fun,
```

Reply:
I would prefer to keep the current state: your option might look as a forgotten comma